enterprise/server/remote_execution/snaputil/snaputil.go

```diff
@@ -23,22 +23,41 @@ import (
 var EnableLocalSnapshotSharing = flag.Bool("executor.enable_local_snapshot_sharing", false, "Enables local snapshot sharing for firecracker VMs. Also requires that executor.firecracker_enable_nbd is true.")
 var EnableRemoteSnapshotSharing = flag.Bool("executor.enable_remote_snapshot_sharing", false, "Enables remote snapshot sharing for firecracker VMs. Also requires that executor.firecracker_enable_nbd and executor.firecracker_enable_uffd are true.")
 
-func GetArtifact(ctx context.Context, localCache interfaces.FileCache, bsClient bytestream.ByteStreamClient, d *repb.Digest, instanceName string, outputPath string) error {
+// ChunkSource represents how a snapshot chunk was initialized
+type ChunkSource int
+
+const (
+	// ChunkSourceUnmapped means the lazy chunk has not been initialized yet
+	ChunkSourceUnmapped ChunkSource = iota
+	// ChunkSourceHole means the chunk was initialized as a hole - i.e. it started
+	// with empty data, though it may have been written to since
+	ChunkSourceHole
+	// ChunkSourceLocalFile means the chunk was created by splitting a snapshot file on disk
+	// into chunks (i.e. this is the first time we're using this snapshot, and it's
+	// not yet cached)
+	ChunkSourceLocalFile
+	// ChunkSourceLocalFilecache means the chunk was fetched from the local filecache
+	ChunkSourceLocalFilecache
+	// ChunkSourceRemoteCache means the chunk was fetched from the remote cache
+	ChunkSourceRemoteCache
```

Comment:
I set it as a single enum because we won't know the source if it's unmapped. 

I had to add a Hole source because if we're dirtying a Hole and copying the chunk, we need a way to specify that as the source for the copied chunk. Agree that we should filter out Holes and Unmapped chunks for the metrics though